PCSX2-WX: Add a mild overclock option on the slider

ssakash opened the pull request:
**Summary of changes**:

* ~~Generalize the calculation formula instead of manually tuning it just to *work* well for specific games.~~
* Revert the EE cycle rate description to the older state.
* Add a mild overclock option to have the 0 at the middle of the slider.
* Rename some variables as the previous names didn't make any sense.
* Convert unnecessary int data types to bool.
* Only use dark green color for the default cycle rate. using a different color just for a single value (previously for -1) might be confusing for users.

Blackbird88 commented:
Does this have anything to do with **EE Timing Hack** by any chance?

avih commented:
Why?

Blackbird88 commented:
I misread the OP. It doesn't seem to fix games relying on it. Nvm.

avih commented:
> Generalize the calculation formula

Why is this better than before? Does it affect the games differently than before? how did you conclude that the new calculation is better? etc etc etc. This info should be at the commit message too.

This is a meaningful change (assuming it actually changes the behavior, which I think it does), and it really needs more background than just stating that you're changing it.

> Needs testing,
>    Crashes on Batman Begins with -1 value on the slider.
>   Performance on ICO with -1 value on the slider.

No. These are only two sample cases which I accidentally discovered. I never claimed they're the only ones and I'm 100% sure they're not the only ones.

Please leave that one specific calculation as is (for the mildest underclock slider notch). It has years of fine tuning and track record with hundreds of games behind it and proved highly compatible with non negligible speedup.

Elegance and reusability of the code is important, but not at the cost of breaking stuff.

If you really want to change it, test the 100 most popular games with it and let us know how many of them it affected badly. It surely won't cover all games which PCSX2 supports, but it would give us a glimpse at the potential compatibility issues with it.


> Revert the EE cycle rate description to the older state.

Whether or not we should use "change by NN %" or "change to XX %" is an open question at #947 which I said I'll handle myself soon. Unless you missed that or you don't believe that I will, I don't see why you change it before it got settled and agreed upon.

And even if you missed that, I explained at the commit message why I changed it. You shouldn't revert things another developer just did without some discussion, unless it's clearly a bug, and even then, it's best to notify that developer and give him/her the chance to handle it.


In general, please split it into multiple commits where each one does only one meaningful change. Off the top of my head (in no particular order), I think these warrant distinct commits:
- add an overclock notch (with its text etc)
- change the calculation (and the affected texts)
- other gui stuff left like maybe more texts and colors.
- whatever renaming and other non functional changes you want to do, and mark it clearly as no-op.

And seriously, just leave that single calculation for that minimum underclock slider position as is regardless of whatever generic formula you come up with and apply to the other slider positions.

ssakash commented:
> No. These are only two sample cases which I accidentally discovered. I never claimed they're the only ones and I'm 100% sure they're not the only ones.

If you're 100% sure then I guess you have another sample case ? why didn't you mention about it ? (or) are you just making assumptions ?

> Elegance and reusability of the code is important, but not at the cost of breaking stuff.

What do you mean by breaking stuff ? According to my testing, the new code might be likely more reliable and stable compared to the older -1 value. If you're claiming it _breaks stuff_ , could you point out any issues with the -1 value on this branch ?

> Whether or not we should use "change by NN %" or "change to XX %" is an open question at #947 which I said I'll handle myself soon. Unless you missed that or you don't believe that I will, I don't see why you change it before it got settled and agreed upon.

Care to point me where you had mentioned that you would handle it ?

> > And even if you missed that, I explained at the commit message why I changed it. You shouldn't revert things another developer just did without some discussion

I never reverted anything _yet_ on the master branch, this PR was opened to have discussions on the first place.
> In general, please split it into multiple commits where each one does only one meaningful change. 

Fair enough, honestly it's a PITA for myself to split everything but if it is easier for you to review the code then sure ! :)

> And seriously, just leave that single calculation for that minimum underclock slider position as is regardless of whatever generic formula you come up with and apply to the other slider positions.

I'm not really against the older calculation as long as you can actually prove that it's better. In the first place, do we need block size based scaling ``scaleblockcycles_calculation()``  ? (which the -1 value does)
If yes, why do we need it ? does it help any games ?

The only issue was due to the higher reduced average ratio of unscaled/scaled block size which should be fixed with the commits on this PR. If you're highly insistent on having the current calculation code for -1 value then please at least provide a single test cases where it performs better than the -1 value on this current branch. Thanks.

avih commented:
> I'm not really against the older calculation as long as you can actually prove that it's better.

No, it was your job to prove that 90b11b2 doesn't break anything, which you didn't do, and now that I restored the original calculation without taking anything away from what 90b11b2 added, you want to remove this calculation again, and so it's your job again to prove that this new calculation also doesn't break anything.

But I'll make it easier for you. Don't prove it, just leave it as is. Do I really need to repeat my arguments?

If you want to change something, _you_ should know why it's better, period. Otherwise, don't change it.

Why do you insist on changing it anyway? you haven't touched it for 9 months, I added an option which only adds value, and two days later you want to remove it? what's going on?

avih commented:
> Care to point me where you had mentioned that you would handle it ?

https://github.com/PCSX2/pcsx2/pull/947#issuecomment-236376980

ssakash commented:
> If you want to change something, you should know why it's better, period. Otherwise, don't change it.

I see, I guess you also follow the same thing then ? could you exactly point out why you had changed the code to scale based on current block values ? how is it better ? does it improve any games ?
```
else if (cyclerate == -1)
		scale_cycles = (s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7) * s_nBlockCycles / 32;
```

As I had stated earlier, the only issue was due to the higher reduced average ratio of unscaled/scaled block size which should be fixed with the commits on this PR as it properly adds a mild underclock value instead of manipulating the value based on block size.


> Why do you insist on changing it anyway? you haven't touched it for 9 months, I added an option which only adds value, and two days later you want to remove it? what's going on?

Your change doesn't seem to target the source of the issue, that's why I don't like it. Also your code has random values to influence the calculation of the scaled blocks value without any good explanation other than **"`` It works nice with ICO and Batman : Begins!``"**

I know that you had reverted the code since you had assumed that the old one was better and you came to that conclusion since it worked better with two specific games ? that certainly doesn't seem ideal to me.

Either way, the conclusion is that the current code for -1 value at this branch is more general and better than -1 value at master branch which you had recently modified.

You were the one who had reported that 90b11b2 breaks ``ICO`` and ``Batman : Begins`` , could you check whether they still have the issue at this branch ? I don't think they would.

Either way, you had suggested to test 100 games which is an impossible task as I don't own that much games but _might_ be feasible for you as you were the one suggested it. (maybe?)

Though I had managed to test the current 20 games which I have in my possession and every single one of them worked fine,

avih commented:
> I see, I guess you also follow the same thing then ? could you exactly point out why you had changed the code to scale based on current block values ? how is it better ? does it improve any games ?

I do.

As the commit message notes, as well as the comment you removed with this PR, it's because it has years of fine tuning behind it, and it was removed accidentally at 90b11b2 without proper explanation, or "proof" as you like to call it.

I restored that, and added comments why it's important to keep it.

We can go on with this forever. You cannot get off the hook with this one. You changed it without explanation which clearly broke things, I restored it to its proven good calculation., and now you want to change it again.

For the forth/fifth/hundredth time, the point is that it had a good balance between compatibility and speed improvement. Your previous change reduced compatibility because it used more extreme value. I restored it. This PR now changes it to have milder effect, which will reduce it's effect to speed up games.

Just leave it. It worked for years, and it's working now.

avih commented:
> Your change doesn't seem to target the source of the issue, that's why I don't like it. Also your code has random values to influence the calculation of the scaled blocks value without any good explanation other than "It works nice with ICO and Batman : Begins!"

It doesn't have random values. It restores whatever was fine tuned over years to yield good results which you removed without proper explanation. The calculation might look different, but it's identical to what it was before. You're welcome to verify that my calculation is correct, and I'll gladly fix it if it isn't.

ssakash commented:
> It doesn't have random values.

_* looks at code *_
```
scale_cycles = (s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7) * s_nBlockCycles / 32;
```

Yeah, no random values! each and every single values seem to make a lot of sense, especially the ``168``!

>  It restores whatever was fine tuned over years to yield good results which you removed without proper explanation

So you basically reverted the code without knowing how the older code works (or) why it scales based on the block size ? If not, could you explain how the older code is actually better from a theoretical perspective ? both myself and refraction agreed that the block size based scaling on older code was pointless.

So if you strongly feel that the block size based scaling carries some great significance then please provide some factual reasons on why it's better instead of providing any vague statements like ``"It was fine tuned so it should work good!"``. 

Also I did give a proper explanation - It was allowed to scale the blocks consistently instead of scaling it differently if the blocks are at a _specific_ range and it did _help_ games compared to the older behavior.

> This PR now changes it to have milder effect, which will reduce it's effect to speed up games.

If you want, I could still make the effect similar to the older code excluding the range based scaling hack you had added. Is that fine with you ? my only problem is with the ``(s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7)`` part.

avih commented:
Just stop it. Leave it as is.

avih commented:
> my only problem is with the (s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7) part.

Live with it.

refractionpcsx2 commented:
That wasn't very constructive of you avih...

avih commented:
There's a limit to the number of times I'm willing to circle around the exact same spot before deciding that another one is not gonna make any difference.

refractionpcsx2 commented:
But you haven't exactly backed up why the old version is better "it's been tuned for years" (when nobody has touched it) isn't really a good argument.

Either come up with valid reasons and form a constructive argument or find somebody who does have valid reasons.  There's no reason to be a douche and say un-constructive things like "shut up" and "live with it" because you've worn out your "it's been tuned" argument.

avih commented:
I didn't tune it, and I haven't tested with all the games it was tuned with to come up with these values.

When I test stuff, frequently it's with this value. My own experience with it is that it has good value (speedup) and high (absolute or close enough) compatibility rate.

Now, 90b11b2 changed it without stating any kind of argument why it's better. Its main goal was to get rid of that list of random values and replace them with a nicer and more unified calculation.

I noticed recently that some games don't work correctly with the new value. I gave them as examples. I wrote something which actually estimates the effect of the of the slider and fount out that 90b11b2 was far from equivalent to what it replaced (focusing on a specific slider value) making the mildest slider position 60% more extreme.

I then restored that specific calculation as the mildest slider value *without* removing any of the calculations or slider positions which 90b11b2 added. I just added another position, that's it. Just another option to select from, which happens to be identical to one of the calculations which 90b11b2 removed.

So far no one got hurt in any way, right? everyone wins. The users get a new position which is less extreme and more compatible, and the other slider positions which 90b11b2 added are still in place like they were before.

Two days later a PR shows up (this) which wants to remove this new slider position. In fact, not only remove it, but also replace the calculations of other slider positions.

When a developer wants to update the code, thy need to have a good reason for it, right? I have mine: the knowledge that it has been file tuned over a long period of time and my own limited experience with it suggests that the tuning was good and the whatever 90b11b2 added accidentally is less good.

So before you want a proof that the value which I restored is better, please provide a proof that the same value which 90b11b2 uses instead is really better. I haven't seen a single shread of acceptable reasoning for that.

And now again, changing values without actually listing any tests or statistics or anything to support a claim that the new values are better.

But I'm not arguing against the new calculations either. I'm just arguing to keep that one single slider position to have its value with its track record backed somewhat by my own experience.

Do whatever unified calculations you want for the other 6 or 100 slider positions, just please leave this specific one as is. I'm not objecting to the calculations of other positions, and we can have it both ways, but he's actually specifically objecting to have this calculation in place, all this while he goes around and changes calculations of many slider position without any kind of reasoning how it would be better for users - backed by facts, not by the warmth in your heart when the calculations are unified.

There, I went that circle again, and we're exactly where we started.

avih commented:
The bottom line is that we can have both what he wants and also what I wants, which is the situation now.

But what he wants is to have only what he wants.

For me the choice should be easy, let's have both.

For everyone else, well.. everyone should be asking what value did 90b11b2 or this PR (even assuming we end up keeping the behavior of that single slider position which I restored) brings to users, because so far I've seen no description/tests/statistics/guesses or anything else which suggests that it's better for users.

ssakash commented:
> When a developer wants to update the code, thy need to have a good reason for it, right? I have mine: the knowledge that it has been file tuned over a long period of time and my own limited experience with it suggests that the tuning was good and the whatever 90b11b2 added accidentally is less good.

 You had _assumed_ that the older code was better just because it worked better with your favorite games ? And, you had reverted the code without even any sort of discussion with the original author of the code (myself) when you yourself don't understand my change in the first place,

I had explicitly stated this on #947 

> I'm fine with it as long as you open a PR with Needs testing label. IIRC, the increased average worked better with some games compared to the older behavior.

>It'd be nice to compare the older behaviour with the new one on even more testcases. (AKA games)

Even if you missed that, it would make sense to discuss about potential changes which would impact user experience before making them which you didn't do either. You had just commit pushed it just because you thought it was better....

> For everyone else, well.. everyone should be asking what value did 90b11b2 or this PR (even assuming we end up keeping the behavior of that single slider position which I restored) brings to users, because so far I've seen no description/tests/statistics/guesses or anything else which suggests that it's better for users.

I had stated it already lots of times , either way I'll mention it again : It was not just done to have all the code look unified and cool. It was to _**have**_ the ratio of scaling of blocks at a constant rate instead of dynamically manipulating them based on some hackery which the older code did.

**older code calculation**:
```
scale_cycles = (s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7) * s_nBlockCycles / 32;
```

**newer code calculation**:
```
scale_cycles = DEFAULT_SCALED_BLOCKS() * (1 - (cyclerate * 0.25));
```

The newer code allows them to have consistent scaling of the block size to a specific % instead of irrationally/randomly changing them just because the value of current recompiling blocks are at a _specific_ range. I have only testcases that it improved the overclocking positively, you have various test cases for that in [youtube](https://www.youtube.com/watch?v=XNWXIjUve9Q&feature=youtu.be) where people had noticed a better performance in overclocking compared to the older behavior.

However I don't think I had such testcases for underclocking. Though either way I had to change the behavior since removing the range based limitation improved overclocking quite a lot and I was quite convinced that it would also help underclocking. And, eventually I did it and then you came along and reverted the code without fully understanding the issue (or) the previous author's intention.

There you go, I've provided my reasons on why it's *_better_*. Now it's your turn, could you also explain on why having those random values on -1 value is better ? please don't say that it was ``fine tuned`` once again. that's just an other way of saying ``it was hacked up to provide good results at  xyz games``  Just give some reasonable theoretical reason on why block size based hacks are better and I promise you I won't that touch that calculation part, thanks.

**(or)**

To make it easier for you, you could provide a test cases where the block size based scaling (``(s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7)`` part) actually improves something and I'll leave it as it is.

Until now, you haven't even provided a single sensible reason on why we must have it. All you've been saying is ``please, let's have it.`` , ``deal with it`` which doesn't really make you look credible :/

> But I'm not arguing against the new calculations either. I'm just arguing to keep that one single slider position to have its value with its track record backed somewhat by my own experience.

You're wrong, I'm not objecting to the calculation of scaling it by 75% with respect to default. I'm only against the hack ``(s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7)``. And no, I can't just deal with it when someone adds a code which they themselves don't know how it works.

> There, I went that circle again, and we're exactly where we started.

It would help if you don't enter the circle once again, thanks :)

avih commented:
> You had assumed that the older code was better just because it worked better with your favorite games ? you had reverted

I have not reverted anything. After 90b11b2 we had 5 different behaviors with 5 slider positions. After my commit we had 6: the exact 5 we had before it, plus one which 90b11b2 removed and has a good track record.

What have I reverted exactly? you have all the great 5 or whatever behaviors you added, they're all there.

> you have various test cases for that in youtube where people had noticed a better performance in overclocking compared to the older behavior.

Obviously that's possible, because it's more extreme. And it's still there. Well, at least before you change it again in this PR which changes them again.

You change it, you do enough testing or provide enough reasoning to state why it would be overall for users.

But I'm even not asking you to do that. Do whatever you want with as many slider positions you want, spread nicely and evenly with whatever elegant and potentially useful calculations you want. I could argue against it, but I'm not.

My only argument is that the same way I live with your untested calculations on all slider positions, you should live with my proven calculation at one slider position, as ugly and random as it might look to you, because it's one line of code and it doesn't take anything away from everything else you're adding.

ssakash commented:
> I have not reverted anything. After 90b11b2 we had 5 different behaviors with 5 slider positions. After my commit we had 6: the exact 5 we had before it, plus one which 90b11b2 removed and has a good track record.

> What have I reverted exactly? you have all the great 5 or whatever behaviors you added, they're all there.

I'll try to be more clear, on your recent commits you had made two certain changes.

1/ Added a mild underclock option (I'm perfectly fine with it , fixes crash issues)
2/ Added random values to scale based on the range of current blocks value. (doesn't fix anything and looks dubious )

As stated earlier, by adding back ``#2`` you had essentially defeated the purpose of my change in https://github.com/PCSX2/pcsx2/commit/90b11b2fb9e46a73dbeef0f4c09e0e3f5bd10922 for underclock values. Hence it's counter-initiative.

> Obviously that's possible, because it's more extreme. And it's still there. **Well, at least before you change it again in this PR which changes them again**.

Don't worry, I'll make sure to open a thread on forums and inform testers before this gets merged for testing on potential regressions. Thanks for the concern.

> you should live with my proven calculation at one slider position, as ugly and random as it **might look** to you

so it **_might_** look random but they are not random values ? care to explain ?

I did say this on my older post,

```
could you also explain on why having those random values on -1 value is better ? please don't say that it was fine tuned once again. that's just an other way of saying it was hacked up to provide good results at xyz games Just give some reasonable theoretical reason on why block size based hacks are better and I promise you I won't that touch that calculation part, thanks.

(or)

To make it easier for you, you could provide a test case where the block size based scaling ((s_nBlockCycles <= 80 || s_nBlockCycles > 168 ? 5 : 7) part) actually improves something and I'll leave it as it is.
```

But I never got a reply from you about it. why are you so insistent on having that hack part actually ? I told you I'm fine with the calculation itself whereas that hack doesn't help anything. I hope you are constructive on your next reply, thanks.

avih commented:
> I'll try to be more clear, on your recent commits you had made two certain changes.
> 
> 1/ Added a mild underclock option (I'm perfectly fine with it , fixes crash issues)
> 2/ Added random values to scale based on the range of current blocks value. (doesn't fix anything and looks dubious )

Unless my commit was buggy, I only did 1, and the means by which I restored this value is what you call random numbers. They're not random, they restore the same calculation which was for the mildest slider position before 90b11b2  . I didn't intend to touch the overclock behaviors and I didn't intend to touch the existing underclock ones (other than moving them one notch down).

Now, it's possible that I missed something and unintentionally also modified the behaviors which were before my commit (other than just moving them one notch down the slider).

If you could tell me exactly which slider position[s] is not available anymore after my commit, or point to the bug which I introduced, I'll gladly fix that.

ssakash commented:
> Unless my commit was buggy, I only did 1

You did 2. ``#1`` relies on ``#2`` to create a mild underclocking effect which I don't like. You could still add ``#1`` without ``#2``.

> They're not random, they restore the same calculation which was for the mildest slider position before 90b11b2

The older code was full of random values, hence I had removed it. But you have re-introduced them once again so it's counter-initiative to the change done in 90b11b2 

And since you're insisting on having ``#2`` for some reason, you should provide some kind of valid reason on why it's necessary similar to the reasons I had provided for 90b11b2 , thanks.

avih commented:
Just tell me this: Did I removed any of the behaviors which 90b11b2 added?

If no, then why are you so obsessed about removing the one which I added? We can have it both ways, you keep your calculations and I'll keep mine, and users will have another option to choose from, an option which got removed at 90b11b2 .

Everyone wins.

ssakash commented:
> Just tell me this: Did I remove any of the behaviors which 90b11b2 added?

Technically ? yes, you did. you moved the -1 and -2 value calculation code to it's preceding value and added back the older random/hacky -1 calculation code for the new -1 value which didn't follow the same logic used by the other underclock values.

> If no, then why are you so obsessed about removing the one which I added? 

 It's a major obstacle on preventing implementation of a more granular slider. I can also ask the same thing on why you're so obsessed on having ``#2`` ? it's pointless.

> We can have it both ways, you keep your calculations and I'll keep mine,

As I said earlier, I'm in favor of reproducing similar calculation without ``#2`` on the code.

I have asked this over 4 times already and you're still ignoring my question, please explain why is ``#2`` necessary for the -1 slider. (I won't bother asking it anymore if you keep ignoring it)

avih commented:
> Technically ? yes, you did

No, I did not. Slider position -1 had behavior A. Slider position -2 had behavior B. A and B are still there (at different slider positions) therefore behavior was removed, other than moved down a notch.

> It's a major obstacle on preventing implementation of a more granular slider.

Sure, show me the problem once you have the patch and if it's beyond you to solve it, I'll help you with that.

> I have asked this over 4 times already and you're still ignoring my question, please explain why is #2 necessary for the -1 slider. (I won't bother asking it anymore if you keep ignoring it)

But I have answered it gazzilion times already. I did not write the original calculation and cannot explain why this specific calculation works better than others, but my limited experience does support this notion.

That's it, simple. It doesn't mean that other calculation can't work better, it just means that this specific one is known to work well, and it was removed, accidentally or not, at 90b11b2, and I want it back. That's it.

Now, show me how having this calculation as one of the slider positions proves a "major obstacle" to you, and I'll help you go over that obstacle.

ssakash commented:
> Now, show me how having this calculation as one of the slider positions proves a "major obstacle" to you, and I'll help you go over that obstacle.

Fair enough, I'll try to provide the commit for more granular implementation of the slider by tomorrow in this PR. (too busy right now)

Then you'll surely be able to understand on how it is an obstacle ;)

ramapcsx2 commented:
@ssakash You probably know that the cycle calculation isn't accurate on the instruction level and especially on memops. I recommend we focus on improving those first and then come up with a fitting EE over/underclock slider. 

Points of interest are, besides others:
https://github.com/PCSX2/pcsx2/blob/master/pcsx2/R5900OpcodeTables.cpp#L97
https://github.com/PCSX2/pcsx2/blob/master/pcsx2/x86/ix86-32/iR5900-32.cpp#L1328

We need to brainstorm for improvements.

ssakash commented:
FWIW, I wasn't really happy with my more granular implementation of the slider, the calculation was quite inconsistent at most places and the implementation I had used for the blocks scaling code was too _naive_ and the scaling didn't result in a consistent average ratio. So I'm going to delay it's implementation until I get a clear idea on how to handle the calculation code -_-

I guess I'll tidy up this PR later to _just_ add a mild overclock option to balance the slider. about ``-1`` value : I guess there's no harm in satisfying avih :P 

@ramapcsx2 

>  I recommend we focus on improving those first and then come up with a fitting EE over/underclock slider. 

I agree, that would surely help us on providing more accurate over/under clock values. Either way, the cycle counting is quite inaccurate and there are lots of guess based calculations at most places. Accuracy on that would surely be worth tackling :)

ssakash commented:
Updated the PR to only add a mild overclock option on the slider, calculation code for the other values are untouched.

avih commented:
> calculation code for the other values are untouched.

Thanks.

Two comments:

- I changed the text from "by XX%" to "to YY%" for a reason which I explained both at my commit message and at https://github.com/PCSX2/pcsx2/pull/947#issuecomment-236374925 and on other comments on that PR, but I'm not seeing an explanation for you wanting to revert it. I'm still waiting for @refractionpcsx2 to respond to https://github.com/PCSX2/pcsx2/pull/947#issuecomment-236845859 , and while I can live fine with it either way, personally I still think using "to" would be better. So what's the reason for this change? Surely you have an opinion and an explanation why you think "by XX%" is better? If you don't, then I'm not sure why you want to change it, and if you do then it should go to the commit message, and preferably discussed first explicitly since you're reverting what another developer just tried to improve.


- Typos aside, and thanks for catching the "overal" one, naming preferences and coding style (e.g. casting, or using int instead of bool, and possibly others) are many times subjective. If it's not broken and can be followed by most programmers - just as you managed to, I think it's generally not a great idea to rename stuff and modify tiny bits which you otherwise don't really touch. I'm talking about the entire "Minor clean up" commit but IMO that's true in general. Unless it really makes the code better, it pollutes the commits history and makes it harder to track when code and changes were introduced, and their commit messages. All in all, IMHO, typically these kinds of changes hurt more than they help - even when they're clearly better. I'm not going to fight you over it, but I think it's unnecessary (i.e. better without this entire commit) - here and in general.

turtleli commented on pcsx2/x86/ix86-32/iR5900-32.cpp:
```diff
@@ -982,24 +984,26 @@ static u32 scaleblockcycles_calc()
+	unscaled_overall += unscaled;
+	float ratio = static_cast<float>(unscaled_overall) / scaled_overall;
 
 	DevCon.WriteLn(L"Unscaled overal: %d,  scaled overal: %d,  relative EE clock speed: %d %%",
```
`overall` is still incorrectly spelt here.

ssakash commented:
**About first comment -** Personally I don't see the advantage on converting it from ``"by XX%"`` to ``"to YY%"`` in the first place. It'll only cause further confusion for the users compared to the previous description and _might_ mislead them to think that the increase is actually the percentage mentioned.

FWIW, I also have a subjective preference to ``"by XX%"`` which can't really be explained. ``(maybe it makes the increase look not so high ?)``

Either way, pinging @refractionpcsx2 to hear his thoughts on this matter.


**About second commit** - I do understand your concern that it pollutes the ``Git History`` but however I have done similar changes in the past and I don't see why it's a concern only this time ? Other than the obvious typo fixes, I didn't really like the older name for the macro ``DEFAULT_SCALING_UNSANITIZED()`` for example which seemed quite confusing.

Also I feel that using ``int`` instead of ``bool`` is quite confusing as well. why not limit it to bool when you only need to check for true/false events ? I also changed the braces placement on the function to have the placements consistent on the file. About converting C style casts to C++, I have no clue why I did it but that doesn't really matter. Either way the conclusion is - I think it's worth keeping.

avih commented:
> but however I have done similar changes in the past and I don't see why it's a concern only this time

I clearly stated it's true in general and not only specifically for this commit. But I don't follow each commit and haven't noticed it in the past, and now I have noticed it, so I mentioned it.

> I didn't really like the older name for the macro DEFAULT_SCALING_UNSANITIZED()

It's not an "old" name. It's actually a new one. We didn't have this macro a week ago - I added it. I didn't rename it from something else either, it's new.

Sometimes I also don't like the names which other developers use. Naming is often personal preference, but you don't see anyone renaming the names which other devs chose.

I'm not saying it should never happen and obviously sometimes it makes sense, as part of API changes or some rewrites, etc. But otherwise, everyone understands that different people have different preferences and we don't enforce ours on others.

> Also I feel that using int instead of bool is quite confusing as well. why not limit it to bool when you only need to check for true/false events ?

Exactly my point. I didn't say int is more appropriate, just that most developers "get it" also as int, and even if the original author could have made a better choice with bool (and I don't disagree with that), changing it now hurts more than it helps in terms of commits history and ability to track meaningful changes at the code.

Again, this is a general comment: in general, don't do non functional changes on what other devs wrote just because you prefer it slightly differently, and even if they could have done a better choice and they didn't. sometimes it's OK, but mostly it hurts more than it helps. The major concern here is that `git blame` becomes quite less useful with each such non-functional change.

You do notice that such no-op commits with renames or style are relatively rare, on this project or other projects, right? There's a reason for that, and it's not laziness. It's respect for others, and the understanding that such changes make it harder to track where functionality actually changes.

avih commented:
> About first comment - Personally I don't see the advantage on converting it from "by XX%" to "to YY%" in the first place.

So, it seems @refractionpcsx2 doesn't care much either way - https://github.com/PCSX2/pcsx2/pull/947#issuecomment-238077723 , which leaves us to choose between with how it was for years, or the format to which I changed it few days ago, which I thought was an improvement for the reasons I mentioned.

IMO Let's leave it for a while and see if it actually confuses people. I think it won't, but if more people prefer it the old way than the new way, then sure, let's revert it to how it was - but _after_ we actually know that.

ssakash commented:
Well, you obviously know what's better so I let you handle it.

avih commented:
> Well, you obviously know what's better so I let you handle it.

For sure I don't _know_ what's better, especially on the text format thingy which is subjective to a degree, but yeah, I hope I was able to improve it, and if it turns out I was wrong, sure, I wouldn't want to confuse our users any more than you do. We do have this common goal after all.

What with the extra overclock value then? I think it's good both because it gives another less extreme option but also because it makes the slider balanced again with default at the exact center.

Wasn't that the actual main thing you wanted to do with this PR? the rest are fluff really, but this commit actually changes things in a way we both agree would be better for users... don't we?

avih commented:
Also, seems like both you and @gregory38 feel that red is better for the -1 slider position so we should probably do that too. If you don't do it at this PR then I can do it myself later too. Up to you.

ssakash commented:
Updated the PR with the suggested changes. 

However I still feel that the ``int -> bool`` , braces placement consistency and the typo fixes should still be included (the whole cleanup commit) , any opinion ? I'm aware of the drawbacks but I still think it's worth it.

avih commented:
> However I still feel that the int -> bool , braces placement consistency and the typo fixes should still be included (the whole cleanup commit) , any opinion ? I'm aware of the drawbacks but I still think it's worth it.

I can live with that.

But in the future, please try to avoid such no-op renames and re-formats unless you think it _really_ makes the code better. There's really high value to being able to track real code and functionality changes without having to "bypass" such no-op changes.

Merge then once Appveyor is green?

ssakash commented:
> But in the future, please try to avoid such no-op renames and re-formats unless you think it really makes the code better. There's really high value to being able to track real code and functionality changes without having to "bypass" such no-op changes.

Right, I'll try to avoid such changes in the future.

> Merge then once Appveyor is green?

Fine by me :+1:

avih commented:
Thanks :)
